src/bio/bio_buffer.c

```diff
@@ -629,18 +629,30 @@ dma_map_one(struct bio_desc *biod, struct bio_iov *biov,
 static void
 rw_completion(void *cb_arg, int err)
 {
-	struct bio_desc *biod = cb_arg;
+	struct bio_desc 	*biod = cb_arg;
+	struct media_error_msg	*mem = NULL;
 
 	ABT_mutex_lock(biod->bd_mutex);
 
 	D_ASSERT(biod->bd_inflights > 0);
 	biod->bd_inflights--;
-	if (biod->bd_result == 0 && err != 0)
+
+	if (biod->bd_result == 0 && err != 0) {
 		biod->bd_result = daos_errno2der(-err);
+		D_ALLOC_PTR(mem);
+		if (mem == NULL)
+			goto skip_media_error;
```

Comment:
wakeup is missed.

Reply:
good catch, will move skip_media_error location